File: app/src/main/java/com/victoria/bleled/common/receiver/SmsReceiver.java
package com.victoria.bleled.common.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.telephony.SmsMessage;

import androidx.localbroadcastmanager.content.LocalBroadcastManager;

import com.victoria.bleled.common.Constants;

import timber.log.Timber;

/**
 * Created by HappyMario on 1/12/2018.
 */

public class SmsReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        if (intent.getAction().equals("android.provider.Telephony.SMS_RECEIVED")) {
            Bundle bundle = intent.getExtras();           //---get the SMS message passed in---
            SmsMessage[] msgs = null;
            String msg_from;
            if (bundle != null) {
                //---retrieve the SMS message received---
                try {
                    Object[] pdus = (Object[]) bundle.get("pdus");
                    msgs = new SmsMessage[pdus.length];
                    for (int i = 0; i < msgs.length; i++) {
                        msgs[i] = SmsMessage.createFromPdu((byte[]) pdus[i]);
                        String authNum = msgs[i].getMessageBody().replaceAll("[^0-9]", "");

                        Intent retIntent = new Intent(Constants.BROADCAST_SMS_RECEIVED);
                        retIntent.putExtra(Constants.ARG_TYPE, authNum);
                        LocalBroadcastManager.getInstance(context).sendBroadcast(retIntent);
                    }
                } catch (Exception e) {
                    Timber.e("onReceive - Exception : " + e.getMessage());
                }
            }
        }
    }
}

File: app/src/main/java/com/mario/template/common/receiver/SmsReceiver.kt
package com.mario.template.common.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.telephony.SmsMessage
import androidx.localbroadcastmanager.content.LocalBroadcastManager
import com.mario.template.Constants
import timber.log.Timber

/**
 * Created by HappyMario on 1/12/2018.
 */
class SmsReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action == "android.provider.Telephony.SMS_RECEIVED") {
            val bundle = intent.extras //---get the SMS message passed in---
            var msgs: Array<SmsMessage?>? = null
            var msg_from: String
            if (bundle != null) {
                //---retrieve the SMS message received---
                try {
                    val pdus = bundle["pdus"] as Array<Any>?
                    msgs = arrayOfNulls(pdus!!.size)
                    for (i in msgs.indices) {
                        msgs[i] = SmsMessage.createFromPdu(pdus[i] as ByteArray)
                        val authNum = msgs[i]!!.getMessageBody().replace("[^0-9]".toRegex(), "")
                        val retIntent = Intent(Constants.BROADCAST_SMS_RECEIVED)
                        retIntent.putExtra(Constants.ARG_TYPE, authNum)
                        LocalBroadcastManager.getInstance(context).sendBroadcast(retIntent)
                    }
                } catch (e: Exception) {
                    Timber.e("onReceive - Exception : " + e.message)
                }
            }
        }
    }
}